src/moves.rs

```diff
@@ -3,6 +3,26 @@ use super::hand::Hands;
 use super::judge::{Judge, Outcome};
 use crate::error::GamePlayError;
 
+#[derive(Debug, PartialEq)]
+pub enum ChangeAction {
+    Added,
+    Swapped,
+    Defeated,
+    Truncated,
+}
```

Comment:
We should also expand this to cover states like `WonAttack`, `WonDefense`... Anything we might want to highlight in the UI. (e.g. highlighting all words that contributed to an attack would be cool)

Reply:
Good idea. I'd like to see which words lost in a defence, and perhaps whether they were invalid or too short.

Reply:
Ooh yes that would be good extra data